fw/rbcx-coprocessor/src/DebugLink.cpp

```diff
@@ -83,6 +110,71 @@ extern "C" int _write(int fd, char* data, int len) {
     return res;
 }
 
+static void debugLinkHandleCommandISR(const char* cmd) {
+    if (strcmp(cmd, "help") == 0) {
+        printf("Available commands: \n"
+               "  calibrate power VCC_MV BATTERY_MID_MV VREF_33V_MV "
+               "TEMPERATURE_C\n");
+        return;
+    }
+
+    if (strncmp(cmd, "calibrate ", 4) == 0) {
+        cmd += 10;
+
+        if (strncmp(cmd, "power ", 6) == 0) {
+            cmd += 6;
+
+            CoprocReq req = {
+                .which_payload = CoprocReq_calibratePower_tag,
+            };
+
+            if (sscanf(cmd, "%lu %lu %lu %lu",
+                    &req.payload.calibratePower.vccMv,
+                    &req.payload.calibratePower.battMidMv,
+                    &req.payload.calibratePower.vRef33Mv,
+                    &req.payload.calibratePower.temperatureC)
+                != 4) {
+                printf("Invalid parameters!\n");
+                return;
+            }
+
+            dispatcherEnqueueRequest(req);
+            return;
+        }
+    }
+
+    printf("Invalid command.\n");
+}
```

Comment:
Tento kód bych nejradši viděl mimo ISR

Reply:
Takže ideálně task, co blokuje forever na frontě

Reply:
To jsem chtěl původně, ale řádek může mít jakoukoliv délku, takže by to musela být něco jako fronta `char[128]`, nebo dynamická alokace, což mi teda přijde pomalejší než tohle.

Reply:
MessageBuf je dobry reseni